### DISCUSSION TITLE
Cannot import LayerMap

### DISCUSSION THREAD

**Discussion by Hohniker** [gdsfactory/gdsfactory#2120] [Q&A] [answered]
Ran this import statement from an example from gdsfactory documentation

from gdsfactory.technology import (
    LayerViews,
    lyp_to_dataclass,
    LayerMap,
)

Output
ImportError: cannot import name 'LayerMap' from 'gdsfactory.technology' 

It doesn't work in either the latest version of gdsfactory as of 9/23/23 or an older version that was installed 4/1/23

**Comment by joamatab** [comment]
You are probably using an old version of gdsfactory

Can you show what you get when you do?

```python

from gdsfactory.technology import LayerViews, lyp_to_dataclass, LayerMap

from gdsfactory.config import print_version_plugins

print_version_plugins()

```

**Reply by Hohniker** [reply]
Brand new installation in virtual env.

When I try the first line, I get:
ImportError: cannot import name 'LayerMap' from 'gdsfactory.technology'

The second line results in

python version = 3.11.0
gdsfactory version = 7.3.5

![image](https://github.com/gdsfactory/gdsfactory/assets/44915520/0e873ad1-3f6d-40a4-8a7a-30fbf9ee0192)

**Comment by joamatab** [comment, marked as answer]
yes, LayerMap was added on version 7.4.6. [see](https://gdsfactory.github.io/gdsfactory/CHANGELOG.html)

the documentation posted online is for 7.8.1

You have 2 options:

- `pip install gdsfactory --upgrade`  and jump to the latest version. You need to add the `--upgrade` to get the latest version
- keep using the old version with no LayerMap

**Reply by Hohniker** [reply]
Ah, that's it!

When I did the fresh install in a new env I didn't run Anaconda Prompt as an administrator and so it had installed an older version. I just took for granted that I had the latest and greatest.

**Comment by SadiMJawadAhsan** [comment]
I have a question here. I am facing the same problem importing LayerMap. But I ran anaconda prompt as administrator and did "pip install gdsfactory --upgrade". The issue didn't get resolved. Each time I do  "pip install gdsfactory --upgrade" it installs version 6.116.0. I am not sure why

**Reply by joamatab** [reply]
You are probably not using python 3.10 or 3.11 as indicated in the instructions
